components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -919,4 +921,30 @@ private AuthenticationResult getAuthenticationResult(HttpServletRequest request,
         return authenticationResult;
     }
 
+    private void setAuthenticatedIDPByApps(SessionContext sessionContext, String applicationName) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Getting shared authenticatedIdpData from session context and setting it for "
+                    + "application: " + applicationName);
+        }
+        Map<String, AuthenticatedIdPData> authenticatedIdPDataMap = new HashMap<>();
+        for (Map.Entry<String, AuthenticatedIdPData> entry : sessionContext.getAuthenticatedIdPs().entrySet()) {
```

Comment:
as discussed offline, we will not need this loop anymore

Reply:
Made AuthenticatedIdpData as cloneable and constructed a map of cloned authenticatedIdPData. (<"GOOGLEIDP", object_of_authenticatedidp>)